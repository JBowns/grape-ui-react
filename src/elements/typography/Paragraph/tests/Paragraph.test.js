import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { Paragraph } from '..';
import 'jest-styled-components';

describe('Paragraph Component base', () => {
  it('should return a Paragraph object with base styling', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Paragraph>Lorem Ipsum</Paragraph></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Paragraph Component specified styles', () => {
  it('should return a Paragraph object with a green color', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Paragraph color="green">Lorem Ipsum</Paragraph></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with ellipsis', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Paragraph ellipsis>Lorem Ipsum</Paragraph></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with a serif font-family', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Paragraph fontFamily="serif">Lorem Ipsum</Paragraph></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with a bold font-weight', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Paragraph fontWeight="bold">Lorem Ipsum</Paragraph></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with font-style: italic', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Paragraph italic>Lorem Ipsum</Paragraph></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with specified kerning', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Paragraph kerning="1px">Lorem Ipsum</Paragraph></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with font-size: lg', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Paragraph lg>Lorem Ipsum</Paragraph></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with font-size: sm', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Paragraph sm>Lorem Ipsum</Paragraph></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with text-align: right', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Paragraph textAlign="right">Lorem Ipsum</Paragraph></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with a lead paragraph stlying', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Paragraph lead>Lorem Ipsum</Paragraph></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with text-decoration: underline', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Paragraph textDecoration="underline">Lorem Ipsum</Paragraph></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Paragraph object with custom margins', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Paragraph mb={1} mt={1}>Lorem Ipsum</Paragraph></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
