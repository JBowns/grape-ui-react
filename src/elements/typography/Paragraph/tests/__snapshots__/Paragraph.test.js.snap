// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph Component base should return a Paragraph object with base styling 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 1rem;
  margin-top: 0;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component specified styles should return a Paragraph object with a bold font-weight 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: bold;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 1rem;
  margin-top: 0;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component specified styles should return a Paragraph object with a green color 1`] = `
.c0 {
  color: hsl(122.4,39.4%,49.2%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 1rem;
  margin-top: 0;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component specified styles should return a Paragraph object with a lead paragraph stlying 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1.2rem;
  font-weight: 300;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 1rem;
  margin-top: 0;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component specified styles should return a Paragraph object with a serif font-family 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: "Palatino Linotype",Palatino,Palladio,"URW Palladio L","Book Antiqua",Baskerville,"Bookman Old Style","Bitstream Charter","Nimbus Roman No9 L",Garamond,"Apple Garamond","ITC Garamond Narrow","New Century Schoolbook","Century Schoolbook","Century Schoolbook L",Georgia,serif;
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 1rem;
  margin-top: 0;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component specified styles should return a Paragraph object with custom margins 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 4px;
  margin-top: 4px;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component specified styles should return a Paragraph object with ellipsis 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 1rem;
  margin-top: 0;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component specified styles should return a Paragraph object with font-size: lg 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1.2rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 1rem;
  margin-top: 0;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component specified styles should return a Paragraph object with font-size: sm 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 1rem;
  margin-top: 0;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component specified styles should return a Paragraph object with font-style: italic 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  font-style: italic;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 1rem;
  margin-top: 0;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component specified styles should return a Paragraph object with specified kerning 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 1rem;
  margin-top: 0;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component specified styles should return a Paragraph object with text-align: right 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: right;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 1rem;
  margin-top: 0;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph Component specified styles should return a Paragraph object with text-decoration: underline 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  margin-bottom: 1rem;
  margin-top: 0;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;
